source/Octopus.Tentacle/Kubernetes/KubernetesJobMonitor.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using k8s;
+using k8s.Models;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Contracts;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public interface IKubernetesJobMonitor
+    {
+        Task StartAsync(CancellationToken token);
+    }
+
+    public interface IKubernetesJobStatusProvider
+    {
+        JobStatus? TryGetJobStatus(ScriptTicket scriptTicket);
+    }
+
+    public class KubernetesJobMonitor : IKubernetesJobMonitor, IKubernetesJobStatusProvider
```

Comment:
I have 2 separate interfaces so that one is for starting the monitor, and the other is for retrieving the status. It means you   are getting an interface that represents the action you want, not the underlying class